src/Database/LSMTree/Internal.hs

```diff
@@ -470,9 +481,16 @@ openSession tr hfs hbio dir = do
         contents <- FS.listDirectory hfs activeDirPath
         unless (Set.null contents) $ throwIO (SessionDirMalformed (FS.mkFsErrorPath hfs activeDirPath))
 
-    -- Nothing to check: snapshots are verified when they are loaded, not when a
-    -- session is restored.
-    checkSnapshotsDirLayout = pure ()
+    -- The snapshots directory should only contain directories for named
+    -- snapshots
```

Comment:
Do we really want to verify all this here? Why not stick with the previous approach of verifying snapshot names on load?

Reply:
It seemed a simple enough thing to include, but I can remove it

Reply:
Removed it